edk2toollib/windows/policy/firmware_policy.py

```diff
@@ -0,0 +1,641 @@
+                   SubKeyName=SubKeyName, ValueName=ValueName, Value=Value)
+
+    def Print(self, prefix: str = ''):
+        print('%sRule' % (prefix,))
+        print('%s  RootKey:  %x' % (prefix, self.RootKey))
+        print('%s  SubKeyNameOffset:  %x' % (prefix, self.OffsetToSubKeyName))
+        print('%s  ValueNameOffset:  %x' % (prefix, self.OffsetToValueName))
+        print('%s  ValueOffset:  %x' % (prefix, self.OffsetToValue))
+        self.SubKeyName.Print(prefix=prefix + '  SubKeyName ')
+        self.ValueName.Print(prefix=prefix + '  ValueName ')
+        self.Value.Print(prefix=prefix + '  ')
+
+    def Serialize(self, ruleOut: bytearray, valueOut: bytearray, offsetInVT: int):
+        self.OffsetToSubKeyName = offsetInVT
+        localArray = bytearray()
+        self.SubKeyName.Serialize(valueOut=localArray)
+        valueOut += localArray
+
+        self.OffsetToValueName = self.OffsetToSubKeyName + len(localArray)
+        localArray = bytearray()
+        self.ValueName.Serialize(valueOut=localArray)
+        valueOut += localArray
+
+        self.OffsetToValue = self.OffsetToValueName + len(localArray)
+        localArray = bytearray()
+        self.Value.Serialize(valueOut=localArray)
+        valueOut += localArray
+
+        ruleOut += struct.pack(self.StructFormat, self.RootKey, self.OffsetToSubKeyName,
+                               self.OffsetToValueName, self.OffsetToValue)
+
+
+class PolicyValueType():
+    """ Class for storing, serializing, deserializing, & printing PolicyValue Types"""
+    StructFormat = '<H'
+    StructSize = struct.calcsize(StructFormat)
+
+    POLICY_VALUE_TYPE_STRING = 0
+    POLICY_VALUE_TYPE_BOOL = 1
+    POLICY_VALUE_TYPE_DWORD = 2
+    POLICY_VALUE_TYPE_DWORD_RANGE = 3
+    POLICY_VALUE_TYPE_DWORD_CHOICE = 4
+    POLICY_VALUE_TYPE_QWORD = 5
+    POLICY_VALUE_TYPE_QWORD_RANGE = 6
+    POLICY_VALUE_TYPE_QWORD_CHOICE = 7
+    POLICY_VALUE_TYPE_OPTION = 8
+    POLICY_VALUE_TYPE_MULTI_STRING = 9
+    POLICY_VALUE_TYPE_BINARY = 10
+
+    SupportedValueTypes = {POLICY_VALUE_TYPE_DWORD,
+                           POLICY_VALUE_TYPE_QWORD,
+                           POLICY_VALUE_TYPE_STRING}
+
+    def __init__(self, Type):
+        if Type not in self.SupportedValueTypes:
+            ErrorMessage = ('Unsupported ValueType: %x' % Type)
+            print(ErrorMessage)
+            if STRICT:
+                raise Exception(ErrorMessage)
+
+        self.vt = Type
+
+    """if offset is not specified, stream fs position is at beginning PolicyValueType struct"""
+    @classmethod
+    def FromFileStream(cls, fs: BinaryIO, fsOffset: int = None):
+        if fsOffset:
+            fs.seek(fsOffset)
+        valueType = struct.unpack('<H', fs.read(2))[0]
+        return cls(valueType)
+
+    @classmethod
+    def FromBytes(cls, b: bytes, bOffset: int = 0):
+        valueType = struct.unpack_from('<H', b, bOffset)[0]
```

Comment:
Please use the self.StructFormat if this is truly the same pack string.

Reply:
Good catch, thanks!

Reply:
Addressed in recent commit